### ISSUE TITLE
App with React Native Proximity on Actual Android Device keep on crashing

### ISSUE THREAD

**Issue by tws9797** [Estimote/react-native-proximity#44]
I have installed the react-native-proximity and test on the example as provided recently. The app keep on crashing. 

Android Version: 9
buildToolsVersion = "28.0.3"
minSdkVersion = 18
compileSdkVersion = 28
targetSdkVersion = 28
supportLibVersion = "28.0.0"

These are my codes:

```
"use strict";

import * as RNEP from "@estimote/react-native-proximity";

const startProximityObserver = () => {

  const ESTIMOTE_APP_ID = "my-attendance-4f8";
  const ESTIMOTE_APP_TOKEN = "9f39c19fb0973f109b17df865462b5bc";

  const zone1 = new RNEP.ProximityZone(1, "Purple");
  zone1.onEnterAction = context => {
    console.log("zone1 onEnter", context);
  };
  zone1.onExitAction = context => {
    console.log("zone1 onExit", context);
  };
  zone1.onChangeAction = contexts => {
    console.log("zone1 onChange", contexts);
  };

  const zone2 = new RNEP.ProximityZone(1, "Green");
  zone2.onEnterAction = context => {
    console.log("zone2 onEnter", context);
  };
  zone2.onExitAction = context => {
    console.log("zone2 onExit", context);
  };
  zone2.onChangeAction = contexts => {
    console.log("zone2 onChange", contexts);
  };

  RNEP.locationPermission.request().then(
    permission => {
      console.log(`location permission: ${permission}`);

      if (permission !== RNEP.locationPermission.DENIED) {
        const credentials = new RNEP.CloudCredentials(
          ESTIMOTE_APP_ID,
          ESTIMOTE_APP_TOKEN
        );

        const config = {
          notification: {
            title: "Exploration mode is on",
            text: "We'll notify you when you're next to something interesting.",
            channel: {
              id: "exploration-mode",
              name: "Exploration Mode"
            }
          }
        };

        RNEP.proximityObserver.initialize(credentials, config);
        RNEP.proximityObserver.startObservingZones([zone1, zone2]);
      }
    },
    error => {
      console.error("Error when trying to obtain location permission", error);
    }
  );

};

const stopProximityObserver = () => { RNEP.proximityObserver.stopObservingZones(); };

export {startProximityObserver, stopProximityObserver};

```
There is no error shown. The app just crash as soon as I allow the bluetooth access by pressing allow button. The problem is with this line

`        RNEP.proximityObserver.startObservingZones([zone1, zone2]);
`

When I removed this line of code, the app will stop crashing, however no log will be shown. Is it because of the tag name? Will the app crash if tag name was wrong? Or is there something wrong with this function? Kindly assist me to solve the issue. Thank you. Sorry for any inconveniences caused…

**Comment by chrisspankroy** [comment]
If you attach a debugger to the Android app and get the app to crash, what does the stack trace say? I've been seeing a crash that I can't pin down too and I'm wondering if this is the same thing

**Comment by tws9797** [comment]
https://forums.estimote.com/t/app-with-react-native-proximity-on-actual-android-device-keep-on-crashing/10203/2
The debugger doesn’t said anything at all. It just crashed. I have attached the log and the possible problem on this post that I have made. 
The problem is with this line
   RNEP.proximityObserver.startObservingZones([zone1, zone2]);



发送自 Windows 10 版邮件应用

发件人: Chris Spankroy
发送时间: Tuesday, June 25, 2019 1:59 AM
收件人: Estimote/react-native-proximity
抄送: tws9797; Author
主题: Re: [Estimote/react-native-proximity] App with React Native Proximityon Actual Android Device keep on crashing (#44)

If you attach a debugger to the Android app and get the app to crash, what does the stack trace say? I've been seeing a crash that I can't pin down too and I'm wondering if this is the same thing

**Comment by chrisspankroy** [comment]
Awesome, I took a look at the link you posted and saw the crash log you posted there. I think the problem is that Android 9 (API 28) started requiring a special permission in the manifest (android.permission.FOREGROUND_SERVICE) whenever you wanted to start a foreground service. I believe that's what `startObservingZones` does internally. If that permission isn't declared in your manifest when that function is called on Android 9 or above, the app will crash

**Comment by tws9797** [comment]
Omg. Thanks a lot. It really works. 

发送自 Windows 10 版邮件应用

发件人: Chris Spankroy
发送时间: Tuesday, June 25, 2019 2:39 AM
收件人: Estimote/react-native-proximity
抄送: tws9797; Author
主题: Re: [Estimote/react-native-proximity] App with React Native Proximityon Actual Android Device keep on crashing (#44)

Awesome, I took a look at the link you posted and saw the crash log you posted there. I think the problem is that Android 9 (API 28) started requiring a special permission in the manifest (android.permission.FOREGROUND_SERVICE) whenever you wanted to start a foreground service. I believe that's what startObservingZones does internally. If that permission isn't declared in your manifest when that function is called on Android 9 or above, the app will crash

**Comment by heypiotr** [comment]
Thanks a lot for tracking this down! :heart:

I've added a note about this permission to the README, to the "Background support" section. (Since the foreground service is only being used if you configured the Proximity Observer for background usage.)

**Comment by chrisspankroy** [comment]
Awesome, thanks! I've closed the PR ( #45 )
